### PR TITLE
[Feature] XCframework support

### PR DESCRIPTION
**Pull request by annakocheshkova**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Introduce support for XCFrameworks [AB#79022](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/79022)

Also, build pipelines were updated to run the script which builds XCFrameworks from the build artifacts, and an integration test was added.

### CODE REVIEW

File: AppCenter.xcworkspace/xcshareddata/xcschemes/All iOS Frameworks.xcscheme [hunk 1]
```diff
@@ -124,8 +124,17 @@
       buildConfiguration = "Debug"
       selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
       selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
-      codeCoverageEnabled = "YES"
-      shouldUseLaunchSchemeArgsEnv = "YES">
+      shouldUseLaunchSchemeArgsEnv = "YES"
```
Comment (iamclement) [inline AppCenter.xcworkspace/xcshareddata/xcschemes/All iOS Frameworks.xcscheme:127]: I wonder if the change in this file is necessary, doesn't look like a big deal.

**Comment by annakocheshkova** [comment]
The CI fails on tests, it's also on develop so not related to PR. Probably a temporary failure.

**Comment by annakocheshkova** [comment]
Verified that iOS and macOS extensions work.

File: Scripts/build-tvos-framework.sh [hunk 2]
```diff
@@ -13,6 +13,9 @@ echo "Building ${TARGET_NAME}."
 # The following line create it in the root folder of the current project.
 PRODUCTS_DIR=${SRCROOT}/../AppCenter-SDK-Apple/tvOS
 
+# Dir to gather all frameworks and build it into xcframework.
```
Comment (MatkovIvan) [inline Scripts/build-tvos-framework.sh:16]:
```suggestion
# The directory to gather all frameworks and build it into xcframework.
```

File: Scripts/build-macos-framework.sh [hunk 3]
```diff
@@ -13,6 +13,10 @@ echo "Building ${TARGET_NAME}."
 # The following line create it in the root folder of the current project.
 PRODUCTS_DIR=${SRCROOT}/../AppCenter-SDK-Apple/macOS
 
+# Dir to gather all frameworks and build it into xcframework.
```
Comment (MatkovIvan) [inline Scripts/build-macos-framework.sh:16]:
```suggestion
# The directory to gather all frameworks and build it into xcframework.
```

File: Scripts/build-ios-framework.sh [hunk 4]
```diff
@@ -22,6 +22,9 @@ echo "Building ${TARGET_NAME}."
 # The following line create it in the root folder of the current project.
 PRODUCTS_DIR="${SRCROOT}/../AppCenter-SDK-Apple/iOS"
 
+# Dir to gather all frameworks and build it into xcframework.
```
Comment (MatkovIvan) [inline Scripts/build-ios-framework.sh:25]:
```suggestion
# The directory to gather all frameworks and build it into xcframework.
```

**Pull request merged**